File: app/src/main/java/io/github/ovso/psytest/ui/main/fragment/VideoFragmentPresenter.java
package io.github.ovso.psytest.ui.main.fragment;

import android.os.Bundle;
import io.github.ovso.psytest.data.network.model.SearchItem;

public interface VideoFragmentPresenter {

  void onActivityCreated(Bundle args);

  void onDestroyView();

  void onItemClick(SearchItem data);

  void onLoadMore();

  void onRefresh();

  boolean onOptionsItemSelected(int itemId);

  interface View {

    void setupRecyclerView();

    void refresh();

    void showVideo(String videoId);

    void showYoutubeUseWarningDialog();

    void setLoaded();

    void setupSwipeRefresh();

    void hideLoading();

    void showLoading();

    void setupAdListener();

    void navigateToWeb(String url);
  }
}

File: app/src/main/java/io/github/ovso/psytest/ui/main/fragment/VideoFragmentPresenter.kt
package io.github.ovso.psytest.ui.main.fragment

import android.os.Bundle
import io.github.ovso.psytest.data.network.model.SearchItem

interface VideoFragmentPresenter {

  fun onActivityCreated(args: Bundle)

  fun onDestroyView()

  fun onItemClick(data: SearchItem)

  fun onLoadMore()

  fun onRefresh()

  fun onOptionsItemSelected(itemId: Int): Boolean

  interface View {

    fun setupRecyclerView()

    fun refresh()

    fun showVideo(videoId: String?)

    fun showYoutubeUseWarningDialog()

    fun setLoaded()

    fun setupSwipeRefresh()

    fun hideLoading()

    fun showLoading()

    fun setupAdListener()

    fun navigateToWeb(
      url: String,
      title: String
    )

    fun addEvent()
  }
}
